src/payments/types.rs

```diff
@@ -0,0 +1,686 @@
+use serde::{Deserialize, Serialize};
+use serde_enum_str::{Deserialize_enum_str, Serialize_enum_str};
+use serde_with::skip_serializing_none;
+
+use crate::payer::{AdditionalInfoPayer, Payer};
+
+/// # PaymentUpdateOptions
+/// Struct to use in [`PaymentUpdateBuilder`](crate::payments::PaymentUpdateBuilder)
+///
+/// <https://www.mercadopago.com.br/developers/pt/reference/payments/_payments_id/put>
+#[skip_serializing_none]
+#[derive(Deserialize, Serialize, Debug, Default)]
+pub struct PaymentUpdateOptions {
+    /// It's a boolean field that exists in two-step payments (such as debit cards). In this type of payment, which is done asynchronously, first, the purchase amount is reserved (capture = false). This amount is captured and not immediately debited from the account. When the money is actually transferred to the collector (the recipient of the payment), the capture of the amount is performed (capture = true).
+    pub capture: Option<bool>,
+    /// Payment expiration date. The valid format for the attribute is as follows - "yyyy-MM-dd'T'HH:mm:ssz". For example - 2022-11-17T09:37:52.000-04:00.
+    pub date_of_expiration: Option<String>,
+    pub status: Option<PaymentStatus>,
+    pub transaction_amount: Option<f32>,
+}
+
+/// # PaymentSearchOptions
+/// Struct to use in [`PaymentSearchBuilder`](crate::payments::PaymentSearchBuilder)
+///
+/// <https://www.mercadopago.com.br/developers/pt/reference/payments/_payments_search/get>
+#[skip_serializing_none]
+#[derive(Deserialize, Serialize, Debug, Default, Clone)]
+pub struct PaymentSearchOptions {
+    /// Parameter used for sorting a list of payments.
+    pub sort: Option<PaymentSearchSort>,
+    /// Quantity of payments returned.
+    pub limit: Option<usize>,
+    /// Quantity of payments to skip.
+    pub offset: Option<usize>,
+    /// Sorts the payment in ascending or descending order.
+    pub criteria: Option<PaymentSearchCriteria>,
+    /// It is an external reference for the payment.
+    ///
+    /// It can be, for example, a hash code from the Central Bank, serving as an origin identifier for the transaction.
+    pub external_reference: Option<String>,
+    /// Parameter used to define the search interval for payments.
+    ///
+    /// It is related to `begin_date` and `end_date`
```

Comment:
Esta descrição e absolutamente horrível


```suggestion
    /// Defines which date field Mercado Pago should check when searching using `begin_date` and `end_date`
    /// 
    /// What this means is that you can filter by update, created date or whatever date
```